src/server/gdb_server.c

```diff
@@ -1736,7 +1759,7 @@ static int gdb_step_continue_packet(struct connection *connection,
 static int gdb_breakpoint_watchpoint_packet(struct connection *connection,
 		char const *packet, int packet_size)
 {
-	struct target *target = get_target_from_connection(connection);
+	struct target *target = get_available_target_from_connection(connection);
```

Comment:
As far as hardware breakpoints, is there a small chance that a previously active target might become unavailable, and then a hardware breakpoint might get set on a different hart than a user might've expected (if get_available_target_from_connection() returns one of the other peers in the SMP group rather than the previous active target that became newly-unavailable)?  Maybe that'd be an exceptionally rare scenario anyway, if it were to happen at all?

Reply:
That could happen, but it's OK because when asked to set a hardware breakpoint we always set that same breakpoint on every target in the SMP group. So it doesn't matter which of the targets in that SMP group is returned here as far as the user is concerned.

Reply:
Good info, thanks.  Wasn't aware that HW breakpoints were fanned out to all group members; makes sense